### PR TITLE
Fix some issues with printing float and double

### PR DESCRIPTION
**Pull request by bigdinotech**
-fixes some issues with printing float and double
-limits the number of decimal places printable to 16

### CODE REVIEW

**Comment by bigdinotech** [comment]
@SidLeung @eriknyquist please review

**Comment by SidLeung** [comment]
@eriknyquist @calvinatintel Looks good!

**Comment by calvinatintel** [comment]
Can we make the title a bit more descriptive?

File: cores/arduino/Print.cpp [hunk 1]
```diff
@@ -286,10 +286,23 @@ size_t Print::printLongLong(unsigned long long n, uint8_t base) {
...
 
 
-size_t Print::printFloat(double number, uint8_t digits)
+size_t Print::printFloat(double number, uint16_t decimalPlaces)
 {
-  char str[50];
-
-  dtostrf(number, 0, digits, str);
+    if(decimalPlaces>16)
+        decimalPlaces = 16;
```
Comment (eriknyquist) [inline cores/arduino/Print.cpp]: Magic number; can you stick this in a #define somewhere? i.e. MAX_DTOSTRF_DIGITS or something

File: cores/arduino/Print.cpp [hunk 2]
```diff
@@ -286,10 +286,23 @@ size_t Print::printLongLong(unsigned long long n, uint8_t base) {
...
+    int intValue = (int)number;
+    while(intValue !=0)
+    {
+        intValue /= 10;
+        digits++;
+    }
+    digits += decimalPlaces;
+    if(number < 0)
+        digits++;
+    char str[digits];
```
Comment (eriknyquist) [inline cores/arduino/Print.cpp]: should this be `char str[digits + 1]` (dtostrf also appends a null termination char)?

**Pull request merged**